common/conf/conf.cpp

```diff
@@ -32,6 +59,71 @@ string GetEnv(const string &var_name, const string &default_value) {
 	}
 }
 
+ExpectedOptionValue CmdlineOptionsIterator::Next() {
```

Comment:
Well written! Didn't look too deep, but looks clean and mean, skimmed through it. Structure and organisation is neat. No problem to understand the intention here.

Good job!

Reply:
I'm not too sure about just the name `conf` though :thinking:

Reply:
I don't like too, TBH. But it's only used by the `MenderConfig` class now and wasn't there unit-testing, it would be solely in the `.cpp` file.